tokio/src/runtime/time/mod.rs

```diff
@@ -190,23 +198,28 @@ impl Driver {
         assert!(!handle.is_shutdown());
 
         // Finds out the min expiration time to park.
-        let locks = (0..rt_handle.time().inner.get_shard_size())
-            .map(|id| rt_handle.time().inner.lock_sharded_wheel(id))
-            .collect::<Vec<_>>();
-
-        let expiration_time = locks
-            .iter()
-            .filter_map(|lock| lock.next_expiration_time())
-            .min();
-
-        rt_handle
-            .time()
-            .inner
-            .next_wake
-            .store(next_wake_time(expiration_time));
-
-        // Safety: After updating the `next_wake`, we drop all the locks.
-        drop(locks);
+        let expiration_time = {
+            let mut wheels_lock = rt_handle
+                .time()
+                .inner
+                .wheels
+                .write()
+                .expect("Timer wheel shards poisoned");
+            let expiration_time = (0..rt_handle.time().inner.get_shard_size())
+                .filter_map(|id| {
+                    let wheel = wheels_lock.get_sharded_wheel_mut(id);
+                    wheel.next_expiration_time()
+                })
```

Comment:
Could something like this make more sense?
```
wheels_lock.0.iter_mut()
    .filter_map(|wheel| wheel.get_mut().next_expiration_time())
    .min();
```
This way, we don't need to touch indexes at all.

Reply:
I like this. At least when calculating the min expiration time we indeed don't need the indexes anymore and we can get rid of one function.